### PR TITLE
feat: pruneBranchAfterAutomerge

### LINKED ISSUE #15084: Support skip pruning of branch after auto-merge
### What would you like Renovate to be able to do?

Renovate should provide an option to skip pruning of the branch after auto-merge is completed.

Problem statement: 
Our current CI tool does not have the intelligence to trigger pipelines with variable branches(new PRs)/Poll scm for the same. 
So, we are using a static branch name for dependency updates (All other than major), let's say 'renovate/deps-update'.
Today branch gets removed after auto-merge, which causes our CI server to throw material not found warning.
I tried setting 'pruneStaleBranches' as true, before realizing it has a different function. 
 

### If you have any ideas on how this should be implemented, please tell us here.

We can have a new config option: pruneBranchAfterAutomerge

type: 	boolean
default: 	true
cli:		--prune-branch-after-automerge
env:		RENOVATE_PRUNE_BRANCH_AFTER_AUTOMERGE

This would allow us to skip the below block when false:
https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/update/pr/automerge.ts#L132-L138

I am a bit confused about default presets and config options. 

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by pc-coder**
## Changes

Add a new config option to skip pruning of the branch after auto-merging.

## Context

Closes #15084

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/default.ts [hunk 1]
```diff
@@ -327,6 +327,10 @@ export const presets: Record<string, Preset> = {
     description: 'Raise a PR first before any automerging',
     automergeType: 'pr',
   },
+  pruneBranchAfterAutomerge: {
+    description: 'Prune branch after automerging',
+    enabled: true,
+  },
```
Comment (viceice) [inline lib/config/presets/internal/default.ts:333]: remove, as it's enabled by default

File: lib/workers/repository/update/pr/automerge.ts [hunk 2]
```diff
@@ -129,6 +130,10 @@ export async function checkAutoMerge(
   });
   if (res) {
     logger.info({ pr: pr.number, prTitle: pr.title }, 'PR automerged');
+    if (!pruneBranchAfterAutomerge) {
+      logger.warn('Skipping pruning of merged branch');
```
Comment (viceice) [inline lib/workers/repository/update/pr/automerge.ts:134]:
```suggestion
      logger.info('Skipping pruning of merged branch');
```

File: lib/config/presets/internal/default.ts (hunk 1 above)
Comment (pc-coder) [reply lib/config/presets/internal/default.ts]: done

File: lib/workers/repository/update/pr/automerge.ts (hunk 2 above)
Comment (pc-coder) [reply lib/workers/repository/update/pr/automerge.ts]: done

File: lib/config/options/index.ts [hunk 3]
```diff
@@ -1491,6 +1491,12 @@ const options: RenovateOptions[] = [
     env: false,
     supportedPlatforms: ['github'],
   },
+  {
+    name: 'pruneBranchAfterAutomerge',
+    description: 'Set to true to enable branch pruning after automerging.',
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:1496]:
```suggestion
    description: 'Set to `true` to enable branch pruning after automerging.',
```

@viceice Can we use backticks to get `true` displayed in a monospace font, or do we need to wrap it in a `<code></code>` element?

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -2080,6 +2080,11 @@ Here's an example of how you would define PR priority so that devDependencies ar
 
 The PR title is important for some of Renovate's matching algorithms (e.g. determining whether to recreate a PR or not) so ideally don't modify it much.
 
+## pruneBranchAfterAutomerge
+
+Configure to `false` to disable puring of merged branches after automerge.
+Defaults to `true`.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:2086]:
```suggestion
By default Renovate deletes, or "prunes", the branch after automerging.
Set `pruneBranchAfterAutomerge` to `false` to keep the branch after automerging.
```

How about something like this?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/config/options/index.ts (hunk 3 above)
Comment (pc-coder) [reply lib/config/options/index.ts]: Looking at the existing code, backticks should be enough. Pushed commit for the same.

File: docs/usage/configuration-options.md (hunk 4 above)
Comment (pc-coder) [reply docs/usage/configuration-options.md:2086]: Referred to an existing config for keeping things consistent, the suggested one looks more verbose and understandable. Committing the same.

File: lib/config/presets/internal/default.ts (hunk 1 above)
Comment (viceice) [reply lib/config/presets/internal/default.ts]: please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting and reduce noise.

**Review by viceice — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**